api/src/main/java/org/geysermc/geyser/api/pack/ResourcePack.java

```diff
@@ -59,6 +62,25 @@ public interface ResourcePack {
     @NonNull
     String contentKey();
 
+    /**
+     * Sets the content key of the resource pack. Lack of a content key can be represented by an empty string.
+     */
+    void contentKey(@Nullable String contentKey);
```

Comment:
I'm aware of the current pack codec impl that tries to resolve a key file - but I don't see a reason why a user couldn't provide the key themselves :p

Although yeah, we could probably also make sure it can only be set once